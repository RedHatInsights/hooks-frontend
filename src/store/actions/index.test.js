jest.mock('../../Utilities/notificationsBackendAPI', () => ({
    get: jest.fn(() => Promise.resolve({})),
    create: jest.fn(() => Promise.resolve({})),
    update: jest.fn(() => Promise.resolve({})),
    destroy: jest.fn(() => Promise.resolve({}))
}));

import * as actionGenerator from '.';

describe('fetchEndpoints', () => {
    it('returns a state object', () => {
        expect(actionGenerator.fetchEndpoints(1, 10)).toMatchSnapshot();
    });
});

describe('fetchEndpoint', () => {
    it('returns a state object', () => {
        expect(actionGenerator.fetchEndpoint(1)).toMatchSnapshot();
    });
});

describe('createEndpoint', () => {
    it('returns a state object', () => {
        expect(actionGenerator.createEndpoint({ name: 'Endpoint name' })).toMatchSnapshot();
    });
});

describe('updateEndpoint()', () => {
    it('returns a state object', () => {
        expect(actionGenerator.updateEndpoint(1, { name: 'Endpoint name' })).toMatchSnapshot();
    });
});

describe('toggleEndpoint', () => {
    it('returns a state object', () => {
        expect(actionGenerator.toggleEndpoint(1, true)).toMatchSnapshot();
    });
});

describe('deleteEndpoint', () => {
    it('returns a state object', () => {
        expect(actionGenerator.deleteEndpoint(1, 'Endpoint name')).toMatchSnapshot();
    });
});

describe('newEndpoint', () => {
    it('returns a state object', () => {
        expect(actionGenerator.newEndpoint()).toMatchSnapshot();
    });
});

describe('fetchFilters', () => {
    it('returns a state object', () => {
        expect(actionGenerator.fetchFilters(1)).toMatchSnapshot();
    });
});

describe('fetchApps', () => {
    it('returns a state object', () => {
        expect(actionGenerator.fetchApps()).toMatchSnapshot();
    });
});
