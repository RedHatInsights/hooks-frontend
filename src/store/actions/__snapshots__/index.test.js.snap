// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createEndpoint returns a state object 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "name": "Endpoint name",
    },
    "notifications": Object {
      "fulfilled": Object {
        "title": "Endpoint Endpoint name created",
        "variant": "success",
      },
      "rejected": Object {
        "title": "Failed to create endpoint Endpoint name",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "SUBMIT_ENDPOINT",
}
`;

exports[`deleteEndpoint returns a state object 1`] = `
Object {
  "meta": Object {
    "notifications": Object {
      "fulfilled": Object {
        "title": "Endpoint Endpoint name deleted",
        "variant": "success",
      },
      "rejected": Object {
        "title": "Failed to delete endpoint Endpoint name",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "DELETE_ENDPOINT",
}
`;

exports[`fetchApps returns a state object 1`] = `
Object {
  "meta": Object {
    "notifications": Object {
      "rejected": Object {
        "title": "Failed to load apps",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "FETCH_APPS",
}
`;

exports[`fetchEndpoint returns a state object 1`] = `
Object {
  "meta": Object {
    "notifications": Object {
      "rejected": Object {
        "title": "Failed to load endpoint 1",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "FETCH_ENDPOINT",
}
`;

exports[`fetchEndpoints returns a state object 1`] = `
Object {
  "meta": Object {
    "notifications": Object {
      "rejected": Object {
        "title": "Failed to load endpoints",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "FETCH_ENDPOINTS",
}
`;

exports[`fetchFilter returns a state object 1`] = `
Object {
  "meta": Object {
    "notifications": Object {
      "rejected": Object {
        "title": "Failed to load filter",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "FETCH_FILTER",
}
`;

exports[`newEndpoint returns a state object 1`] = `
Object {
  "type": "NEW_ENDPOINT",
}
`;

exports[`testEndpoint returns a state object 1`] = `
Object {
  "meta": Object {
    "notifications": Object {
      "fulfilled": Object {
        "title": "Test event delivery successful",
        "variant": "success",
      },
      "rejected": Object {
        "title": "Test event delivery failed",
        "variant": "warning",
      },
    },
  },
  "payload": Promise {},
  "type": "TEST_ENDPOINT",
}
`;

exports[`toggleEndpoint returns a state object 1`] = `
Object {
  "meta": Object {
    "notifications": Object {
      "rejected": Object {
        "title": "Failed to toggle endpoint 1",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "SUBMIT_ENDPOINT",
}
`;

exports[`updateEndpoint() returns a state object 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "name": "Endpoint name",
    },
    "notifications": Object {
      "fulfilled": Object {
        "title": "Endpoint Endpoint name updated",
        "variant": "success",
      },
      "rejected": Object {
        "title": "Failed to update endpoint Endpoint name",
        "variant": "danger",
      },
    },
  },
  "payload": Promise {},
  "type": "SUBMIT_ENDPOINT",
}
`;
